client/packages/electron/src/keyboardScanner/README.md

```diff
@@ -0,0 +1,51 @@
+## Keyboard Barcode Scanner
+
+Although our preference is to use USB serial mode for barcode scanners, not all scanners support this mode and only allow keyboard emulation over USB. 
+
+To mimic existing USB serial barcode scanner implementation with keyboard scanners we have to capture keyboard input and pass through through all input that does not come from scanner.
+
+#### What is scanner input ?
+
+Because scanner scans and sends keyboard events very fast, we can idetify scanner input by looking the speed at which keys are entered. Two configurations are used to determine if input is form barcode scanner (at the time of writting these are hardcoded):
+
+`maxMSBetweenScannerKeyInputs` defaults to 50 and `minBarcodeLength` defaults to 5.
+
+Of course it's possible for user to spam keyboard and replicate scanner input, this should be OK because:
+
+* It's highly unlikely
+* Our UI is not really catered for operations at this speed (even without barcode scanner these operations are not really considered, as they are unlikely low probabiliy and low impact)
+* We actually recommend using USB serial scanner mode
+
+#### How does it work ?
+
+During barcode scanning mode (when keyboard scanner is selected), all keyboard input is captured with `window.webContents.on('before-input-event'` and stored in a buffer.
+
+A processor is triggered every PROCESSOR_INTERVAL (50ms), it will try to find scanner input by iterating over buffer and finding input seqences matching scanner input criteria. Any input not matching scanner input criteria will be passed through (as if it wasn't captured).
+
+If sequence input is not `finalised` (still ongoing while processor is triggered), it will be put back into buffer.
+
+Processor is started (setInterval) when scanning operation is started and will be cleared when scanning operation is stopped (clearInterval)
+
+#### Passing through input
+
+Any control keys (keysToPassThrough array) are passed through automatically (when input is captured). Other input that is identified by processor as not scanner input is passed through by manually calling keyDown and char event (apparently that is what [chromium](https://stackoverflow.com/a/53223619) needs to consider it input entered by keyboard). 
+
+Since we are capturing input with `window.webContents.on('before-input-event'` we have to temporary allow this input to go through with `allowEvents` flag. Which is set in the processor when input is identified as not being from scanner.
```

Comment:
:nit 
```suggestion
Since we are capturing input with `window.webContents.on('before-input-event'` we have to temporary allow this input to go through with `allowEvents` flag which is set in the processor when the input is identified as not being from scanner.
```